### ISSUE TITLE
Windows SSM agent session initiation very slow if port 80 egress blocked and S3 log upload configured

### ISSUE THREAD

**Issue by nonspecialist** [aws/amazon-ssm-agent#241]
When initiating an SSM session (either via the CLI or via the console) to a Windows instance (haven't tried against Linux, sorry) running the SSM agent with S3 log upload configured, if there's a security group rule in place which blocks port 80 egress, the session can take several minutes to start up -- this appears as a "hang" after issuing the `aws ssm start-session --target <instance-id>` command.

Logs from the EC2 instance in question (`c:\programdata\Amazon\SSM\Logs\amazon-ssm-agent.log`) show entries like:

```
2019-12-11 14:13:23 INFO [ssm-session-worker] [RoleName-011e8a35a1cd6e285] [DataBackend] [pluginName=Standard_Stream] Error when querying S3 bucket using address http://foo.bar.baz.oif.s3.ap-southeast-2.amazonaws.com. Error details: Failed query S3 using url http://foo.bar.baz.oif.s3.ap-southeast-2.amazonaws.com - Head http://foo.bar.baz.oif.s3.ap-southeast-2.amazonaws.com: dial tcp 52.95.134.26:80: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.. Retrying with the generic regional endpoint s3.amazonaws.com...
```

Unblocking egress port 80 means the session is established instantly.

Either port 80 shouldn't be needed, or shouldn't take so long to time out, or the requirement should be documented

**Comment by cflee** [comment]
I've only used SSM Agent on Linux instances, but have also been observing a 2 min 30 sec delay in starting sessions for quite some time -- I just found that it was fixed in my latest agent upgrade, and bisected it down to being fixed in [2.3.707.0](https://github.com/aws/amazon-ssm-agent/releases/tag/2.3.707.0) and up.

Probably was "Switch the request protocol order for getting S3 Header" in ed954f758754e04c6283581dfe81860411d890d3

**Comment by Suchismita1tutun** [comment]
I have worked on windows bastion host. Only 443 outbound port needs to be opened. It worked fine for me.

**Comment by ranjrish** [comment]
This has been fixed from SSM agent v2.3.707.0 onwards. Please update the ssm to the latest version
